pkg/encoding/v1/encoding.go

```diff
@@ -219,11 +222,81 @@ func (m *Mount) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+type Health struct {
+	ReadinessProbe interface{} `yaml:"readinessProbe,omitempty"`
+	Readiness      *api_v1.Probe
+
+	LivenessProbe interface{} `yaml:"livenessProbe,omitempty"`
+	Liveness      *api_v1.Probe
+}
+
+func interfaceToJSON(i interface{}) interface{} {
```

Comment:
can you please add a comment explaining what this actually does and why it's needed?